### PR TITLE
[spirv] Support multiple reductions fused in single dispatch cases

### PR DESCRIPTION
**Pull request by antiagainst**
This commit reuses passes introdued for LLVMGPU and enables the support for multiple reduction fused in a single dispatch cases. This enables more aggressive fusion in the SPIR-V pipeline.

### CODE REVIEW

File: tests/e2e/regression/BUILD [hunk 1]
```diff
@@ -157,6 +157,27 @@ iree_check_single_backend_test_suite(
...
+    ],
+    driver = "vulkan",
+    tags = [
+        # CUDA cuInit fails with sanitizer on.
+        "noasan",
+        "nomsan",
+        "notsan",
+        "noubsan",
+        "requires-gpu-nvidia",
+    ],
```
Comment (ThomasRaoux) [inline tests/e2e/regression/BUILD]: I don't think you need that?

**Review by MaheshRavishankar — APPROVED** [review]
Nice!

**Review by ThomasRaoux — APPROVED** [review]

File: tests/e2e/regression/BUILD (hunk 1 above)
Comment (antiagainst) [reply tests/e2e/regression/BUILD]: The comment was copy pasted and removed. The sanitiers vulkan side also need them.

**Comment by antiagainst** [comment]
Weird. The tests are passing on my local RTX 2070 Super..

**Comment by antiagainst** [comment]
This is likely a driver bug.. The test also passes on Pixel 6 and RX 7900 XTX. Need to update the driver in CI..

**Comment by antiagainst** [comment]
Dropping the regression test for now and land the rest. I'll update the drivers and then enable the regression test.

**Pull request merged**